README.rst

```diff
@@ -41,7 +41,7 @@ Hello World
    >>> pdf = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
    >>> CLs_obs, CLs_exp = pyhf.infer.hypotest(1.0, [51, 48] + pdf.config.auxdata, pdf, return_expected=True)
    >>> print('Observed: {}, Expected: {}'.format(CLs_obs, CLs_exp))
-   Observed: [0.05290116], Expected: [0.06445521]
+   Observed: [0.05290051], Expected: [0.0644532]
```

Comment:
why does this refactor change results?

Reply:
If I'm honest, I don't know. The best fit pars are the same before and after refactor


```python
#before
>>> pyhf.infer.mle.fit([51, 48] + pdf.config.auxdata, pdf)
array([0.        , 1.0030512 , 0.96266961])

#after
>>> pyhf.infer.mle.fit([51, 48] + pdf.config.auxdata, pdf)
array([0.        , 1.0030512 , 0.96266961])
```

Reply:
casting

```python
#before
>>> pdf = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
>>> pyhf.infer.mle.fit([51, 48] + pdf.config.auxdata, pdf).tolist()
[0.0, 1.00305120262146, 0.9626696070391361]

#after
>>> pdf = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
>>> pyhf.infer.mle.fit([51, 48] + pdf.config.auxdata, pdf).tolist()
[0.0, 1.00305120262146, 0.9626696070391361]
```

Reply:
This is split out into #970.